addons/mail/static/src/components/message_action_list/message_action_list.xml

```diff
@@ -2,26 +2,10 @@
-                            'o_MessageActionList_actionStar_active': messageActionList.message.isStarred,
-                            'fa fa-lg fa-star': messageActionList.message.isStarred,
-                            'fa fa-lg fa-star-o': !messageActionList.message.isStarred,
-                        }" role="button" tabindex="0" aria-label="Mark as Todo" t-att-aria-pressed="messageActionList.message.isStarred ? 'true' : 'false'" title="Mark as Todo" t-on-click="messageActionList.onClickToggleStar"/>
-                    <span t-if="messageActionList.hasReplyIcon" class="o_MessageActionList_action o_MessageActionList_actionReply fa fa-lg fa-reply cursor-pointer" t-attf-class="{{ messaging.device.isSmall ? 'px-2 py-3' : 'px-1 py-2' }}" title="Reply" role="button" tabindex="0" aria-label="Reply" t-on-click="messageActionList.onClickReplyTo"/>
-                    <span t-if="messageActionList.hasMarkAsReadIcon" class="o_MessageActionList_action o_MessageActionList_actionMarkRead fa fa-lg fa-check cursor-pointer" t-attf-class="{{ messaging.device.isSmall ? 'px-2 py-3' : 'px-1 py-2' }}" title="Mark as Read" role="button" tabindex="0" aria-label="Mark as Read" t-on-click="messageActionList.onClickMarkAsRead"/>
-                    <span t-if="messageActionList.message.canBeDeleted" class="o_MessageActionList_action o_MessageActionList_actionEdit fa fa-lg fa-pencil cursor-pointer" t-attf-class="{{ messaging.device.isSmall ? 'px-2 py-3' : 'px-1 py-2' }}" title="Edit" role="button" tabindex="0" aria-label="Edit" t-on-click="messageActionList.onClickEdit"/>
-                    <!-- If there is no compact icon, delete will be the last button of the action list, a different styling will be applied. -->
-                    <span t-if="messageActionList.message.canBeDeleted" class="o_MessageActionList_action o_MessageActionList_actionDelete fa fa-lg fa-trash cursor-pointer" t-att-class="{
-                            'px-2 py-3': messageActionList.hasCompactIcon and messaging.device.isSmall,
-                            'px-1 py-2': messageActionList.hasCompactIcon and !messaging.device.isSmall,
-                            'ps-2 py-3 pe-3': !messageActionList.hasCompactIcon and messaging.device.isSmall,
-                            'ps-1 py-2 pe-2': !messageActionList.hasCompactIcon and !messaging.device.isSmall,
-                        }" title="Delete" role="button" tabindex="0" aria-label="Delete" t-on-click="messageActionList.onClickDelete"/>
+            <div class="o_MessageActionList d-flex rounded-1 bg-white shadow-sm" t-att-class="{ 'o-isDeviceSmall': messaging.device.isSmall }" t-attf-class="{{ className }}" t-ref="root">
+                <t t-foreach="messageActionList.messageActionViews" t-as="messageActionView" t-key="messageActionView.localId">
+                    <MessageActionView t-if="!messageActionView.isDisabled" record="messageActionView"/>
```

Comment:
This condition is strange, view shouldn't exist if it shouldn't be displayed.

Reply:
I think the issue comes because we are also actually lacking a `MessageAction` model that is related to state of the message, and not to what is actually displayed?

Reply:
yeah, I think the problem is it's hard to define *the real actions the message had* and *the actions to be displayed in view* inside the view model. that's why I use a field to say if it is showed up.